File: mcap-web/src/main/resources/static/js/controller/function/sysResourceController.js
//控制层
app.controller("sysResourceController", function ($scope, $controller, sysResourceService) {
    //继承
    $controller("baseController", {$scope: $scope});

    //初始化
    $scope.initForAdd = function () {
        $scope.sysResource = {};
    }
    $scope.initForUpdate = function (id) {
        sysResourceService.queryById(id).success(function (data) {
            $scope.sysResource = data;
        });
    }

    //crud
    //查询,初始化查询条件为空
    $scope.condition = {};
    $scope.query = function (pageNum, pageSize) {
        sysResourceService.query(pageNum, pageSize, $scope.condition).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //保存
    $scope.save = function () {
        sysResourceService.save($scope.sysResource).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //批量删除
    $scope.delete = function () {
        $scope.deleteConfirm(deleteRows);
    }
    function deleteRows(){
        sysResourceService.delete($scope.selectedIds).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
});

File: halcyon-base/src/main/java/com/marshal/halcyon/base/function/controller/SysResourceController.java
package com.marshal.halcyon.base.function.controller;


import com.marshal.halcyon.base.function.entity.SysResource;
import com.marshal.halcyon.base.function.service.SysResourceService;
import com.marshal.halcyon.core.controller.BaseController;
import com.marshal.halcyon.core.entity.ResponseData;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * Created with IntelliJ IDEA.
 * author: Marshal
 * Date: 2018/11/5
 * Time: 22:06
 * Description:
 */
@RestController
@RequestMapping("/sys/resource")
public class SysResourceController extends BaseController {

    @Autowired
    SysResourceService sysResourceService;

    @RequestMapping("/query")
    public ResponseData query(@RequestBody SysResource condition, int pageNum, int pageSize) {
        List<SysResource> list = sysResourceService.select(condition, pageNum, pageSize);
        return new ResponseData(list);
    }

    @RequestMapping("/save")
    public ResponseData save(@RequestBody SysResource sysResource) {
        if (getValidator().hasError(sysResource)) {
            return new ResponseData(false, getValidator().getErrors(sysResource));
        }
        sysResourceService.save(sysResource);
        return new ResponseData(true, "保存成功");
    }

    @RequestMapping("/delete")
    public ResponseData delete(@RequestParam("selectedIds") Long[] selectedIds) {
        sysResourceService.batchDelete(selectedIds);
        return new ResponseData(true, "删除成功");
    }

    @RequestMapping("/queryById")
    public SysResource queryById(@RequestParam Long id) {
        return sysResourceService.selectByPrimaryKey(id);
    }

    @RequestMapping("/getResourceOptions")
    public List<Map> getFunctionOptions() {
        return sysResourceService.getResourceOptions();
    }
}
